src/Juvix/Core/IR/Extension.hs

```diff
@@ -0,0 +1,97 @@
+-- |
+-- Extensions to the IR abstract syntax.
+module Juvix.Core.IR.Extension
+where
+
+import Juvix.Library
+import Data.Kind (Type, Constraint)
+
+
+-- | Marker type for having no extensions
+data NoExt
+
+-- | Extensions for terms and eliminations.
```

Comment:
Probably best to put this documentation also at the top show it shows up in the documentation generation?